sdk/testing/server/fixture.go

```diff
@@ -310,9 +310,6 @@ func (c *kcpServer) Run(t TestingT) error {
 }
 
 func (c *kcpServer) Stop() {
-	c.lock.Lock()
-	defer c.lock.Unlock()
```

Comment:
Just a minor q, why are we removing this?

Reply:
That was this commit: https://github.com/kcp-dev/kcp/pull/3372/commits/206456112353c6dfd873ba1a189898b0f826b990
The lock is removed because `.cancel` is locking so calling `.Stop` was actually producing a deadlock :D